### PR TITLE
Implement dot product and try adding test

### PR DESCRIPTION
**Pull request by cgccuser**
Implements dot product. Merge after #4.

### CODE REVIEW

File: compiler/src/main/kotlin/io/github/seggan/myxal/compiler/jvm/JvmCompiler.kt [hunk 1]
```diff
@@ -3,7 +3,7 @@ package io.github.seggan.myxal.compiler.jvm
 import io.github.seggan.myxal.compiler.Element
 import io.github.seggan.myxal.compiler.ICompiler
 import io.github.seggan.myxal.compiler.MyxalCompileException
-import io.github.seggan.myxal.compiler.jvm.msplit.SplitMethod
+// import io.github.seggan.myxal.compiler.jvm.msplit.SplitMethod
```
Comment (Seggan) [inline compiler/src/main/kotlin/io/github/seggan/myxal/compiler/jvm/JvmCompiler.kt:6]: Can remove this comment

File: compiler/src/main/kotlin/io/github/seggan/myxal/compiler/jvm/wrappers/MyxalMethod.kt [hunk 2]
```diff
@@ -1,6 +1,6 @@
 package io.github.seggan.myxal.compiler.jvm.wrappers
 
-import io.github.seggan.myxal.compiler.jvm.msplit.SplitMethod
+// import io.github.seggan.myxal.compiler.jvm.msplit.SplitMethod
```
Comment (Seggan) [inline compiler/src/main/kotlin/io/github/seggan/myxal/compiler/jvm/wrappers/MyxalMethod.kt:3]: Same here

File: runtime/src/main/kotlin/io/github/seggan/myxal/runtime/RuntimeMethods.kt [hunk 3]
```diff
@@ -256,6 +256,12 @@ fun divFive(obj: Any): Any = divisibleBy(obj, 5)
 
 fun divThree(obj: Any): Any = divisibleBy(obj, 3)
 
+fun dotProduct(stack: ProgramStack): Any {
+    val rhs = stack.pop()
```
Comment (Seggan) [inline runtime/src/main/kotlin/io/github/seggan/myxal/runtime/RuntimeMethods.kt:260]: Please use `a` and `b` for consistency

File: runtime/src/test/kotlin/io/github/seggan/myxal/runtime/ListTests.kt [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+                "-8foofoo",
+                dotProduct(
+                        ProgramStack(
+                                listOf(
+                                        JyxalList.create(BigComplex.valueOf(-2), "foo"),
+                                        JyxalList.create(
+                                                BigComplex.valueOf(4),
+                                                BigComplex.valueOf(2)
+                                        )
+                                )
```
Comment (Seggan) [inline runtime/src/test/kotlin/io/github/seggan/myxal/runtime/ListTests.kt:22]:
```suggestion
                                ).jyxal()
```

**Review by Seggan — CHANGES_REQUESTED** [review]

**Comment by cgccuser** [comment]
@Seggan I made PR 2 before #4, so it hasn't caught up yet. I'll fix that stuff once #4 is merged.

**Comment by cgccuser** [comment]
Look at #5 instead

**Pull request closed without merging**